src/utils.rs

```diff
@@ -44,3 +44,8 @@ impl<T> serde::Serialize for SerializablePhantom<T> {
         serializer.serialize_unit_struct(std::any::type_name::<T>())
     }
 }
+
+/// Calculates squared Euclidean distance between two vectors.
+pub fn squared_euclidean(a: &[f64], b: &Vec<f64>) -> f64 {
+    a.iter().zip(b).map(|(p, q)| (p - q).powf(2.0)).sum::<f64>()
```

Comment:
Using `.powi(2)` instead of `.powf(2.0)` is generally the better option for integer exponents.